### PR TITLE
07 22 search all

### PR DESCRIPTION
**Pull request by yiwenwang-ms**
Update search all

### CODE REVIEW

File: api-reference/beta/resources/searchrequest.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ The JSON blob contains the types of resources expected in the response, the unde
 |from|Int32|Specifies the offset for the search results. Offset 0 returns the very first result. Optional.|
 |query|[searchQuery](searchquery.md)|Contains the query terms. Required.|
 |queryAlterationOptions|[searchAlterationOptions](searchalterationoptions.md)|Provides query alteration options formatted as a JSON blob that contains two optional flags related to spelling correction. Optional. |
+|region|string|Specific property for application permission, represent the location of the  storage. |
```
Comment (Lauragra) [inline api-reference/beta/resources/searchrequest.md:38]: Please clarify this description. What storage are you referring to, and what is being stored? Is this property only used in application permission scenarios?

File: api-reference/beta/resources/searchrequest.md [hunk 2]
```diff
@@ -84,6 +85,8 @@ The following is a JSON representation of the resource.
 - [Use aggregations](/graph/search-concept-aggregation) to refine search results
 - [Use display layout](/graph/search-concept-display-layout)
 - [Enable spell corrections](/graph/search-concept-speller) in search results
+- [Search SharePoint content with Application Permission](/graph/search-concept-searchAll)
+- [Interleave file and connector search results](/graph/search-concept-connector-interleaving)
```
Comment (Lauragra) [inline api-reference/beta/resources/searchrequest.md:89]: What are you linking to here? I don't see this file in the content set.

File: api-reference/beta/resources/searchrequest.md (hunk 1 above)
Comment (Lauragra) [inline api-reference/beta/resources/searchrequest.md:38]: Please clarify this description. How is this property related to application permissions? What is being stored in the storage location? The query terms?

File: changelog/Microsoft.Search.json [hunk 3]
```diff
@@ -1197,6 +1197,40 @@
       "CreatedDateTime": "2022-04-28T00:00:00",
       "WorkloadArea": "Search",
       "SubArea": ""
+    },
+    {
+      "ChangeList": [
+        {
+          "Id": "d7e73514-9dad-4c2f-8762-786788ac6984",
+          "ApiChange": "Enumeration",
+          "ChangedApiName": "searchContent",
```
Comment (Lauragra) [inline changelog/Microsoft.Search.json:1206]: I don't see any associated documentation update for this enumeration. Please update the docs accordingly.

File: changelog/Microsoft.Search.json [hunk 4]
```diff
@@ -1197,6 +1197,40 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **searchContent** enumeration type.",
+          "Target": "searchContent"
+        },
+        {
+          "Id": "d7e73514-9dad-4c2f-8762-786788ac6984",
+          "ApiChange": "Property",
+          "ChangedApiName": "sharePointOneDriveOptions",
```
Comment (Lauragra) [inline changelog/Microsoft.Search.json:1214]: Please update the searchRequest resource topic to include this property.

File: changelog/Microsoft.Search.json [hunk 5]
```diff
@@ -1197,6 +1197,40 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the `sharePointOneDriveOptions` property to the [searchRequest](https://docs.microsoft.com/en-us/graph/api/resources/searchRequest?view=graph-rest-beta) resource.",
+          "Target": "searchRequest"
+        },
+        {
+          "Id": "d7e73514-9dad-4c2f-8762-786788ac6984",
+          "ApiChange": "Resource",
+          "ChangedApiName": "sharePointOneDriveOptions",
+          "ChangeType": "Addition",
+          "Description": "Added the [sharePointOneDriveOptions](https://docs.microsoft.com/en-us/graph/api/resources/sharePointOneDriveOptions?view=graph-rest-beta) resource type.",
```
Comment (Lauragra) [inline changelog/Microsoft.Search.json:1224]: Please add a resource topic for this new resource.

File: changelog/Microsoft.Search.json [hunk 6]
```diff
@@ -1197,6 +1197,40 @@
...
+          "Target": "sharePointOneDriveOptions"
+        }
+      ],
+      "Id": "d7e73514-9dad-4c2f-8762-786788ac6984",
+      "Cloud": "Prod",
+      "Version": "beta",
+      "CreatedDateTime": "2022-09-07T06:50:44.0853657Z",
+      "WorkloadArea": "People and workplace intelligence",
```
Comment (Lauragra) [inline changelog/Microsoft.Search.json:1232]: Please change the WorkloadArea value to "Search".

File: concepts/search-concept-searchall.md [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+title: "Search content with application permissions"
+description: "Learn how to use the Microsoft Search API in Microsoft Graph to search SharePoint content using application permissions."
+author: "yiwenwang"
+ms.localizationpriority: medium
+ms.prod: "search"
+---
+
+# Search content with application permissions
+
+This article describes how you can use application permissions with the Microsoft Search API in Microsoft Graph to search SharePoint content. Application permissions are different than delegated permissions in that they enable the application owner to search all content in their SharePoint sites.
```
Comment (Lauragra) [inline concepts/search-concept-searchall.md:11]: I made revisions and edits to this topic; please review the content carefully to verify that it conveys the intended meaning accurately.

File: concepts/search-concept-searchall.md [hunk 8]
```diff
@@ -0,0 +1,272 @@
+---
+title: "Search content with application permissions"
```
Comment (Lauragra) [inline concepts/search-concept-searchall.md:2]: Please add this new concept topic to the toc.yml file in the concepts folder so that it's discoverable in the TOC.

File: concepts/search-concept-searchall.md [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+```
+
+
+## Example 2: Include all private content
+By default, private content search is disabled for customers with application permissions.
+
+To include all private content in a search using application permissions, specify `privateContent` in the **sharePointOneDriveOptions** property.
+
+> [!NOTE]
+> Searching all private content can be an expensive option. When you run an initial search request, if the tenant does not have an active APC stamp, you will need a provision a new APC stamp for your tenant. This can take several hours or days to complete. After the provisioning, the search will work as normal.
```
Comment (Lauragra) [inline concepts/search-concept-searchall.md:145]: Will the reader know what an APC stamp is? Note that we generally spell out acronyms on first mention for clarity, especially if they are likely to be new or unfamiliar to the target audience. It might be helpful here to add a link to admin documentation about APC stamps, for context.

File: api-reference/beta/resources/searchrequest.md (hunk 1 above)
Comment (yiwenwang-ms) [reply api-reference/beta/resources/searchrequest.md]: Thanks Laura for the detailed comment.
I did update below:
1) Updated the description of region, and linked the get the region value to search-concept-searchall.md as there are some samples and also have the approaches to get the region value.
2) Added the region into JSON representation
3) Added application permission in see also
4) Updated the Application permission in the change log.

File: changelog/Microsoft.Search.json (hunk 5 above)
Comment (yiwenwang-ms) [reply changelog/Microsoft.Search.json:1224]: Updated!
1) add sharePointOneDriveOptions in searchRequest and provided the json sample.
2) created new md file for sharePointOneDriveOptions.
3) rename searchContent to includeContent and created new md file for includeContent.

File: concepts/search-concept-searchall.md (hunk 7 above)
Comment (yiwenwang-ms) [reply concepts/search-concept-searchall.md:11]: Thanks for the revision, it looks great to me.

File: concepts/search-concept-searchall.md (hunk 8 above)
Comment (yiwenwang-ms) [reply concepts/search-concept-searchall.md:2]: Done

File: api-reference/beta/resources/searchrequest.md (hunk 2 above)
Comment (yiwenwang-ms) [reply api-reference/beta/resources/searchrequest.md]: Thank you! I removed this one
- [Interleave file and connector search results](/graph/search-concept-connector-interleaving)

File: api-reference/beta/api/search-query.md [hunk 10]
```diff
@@ -25,7 +25,7 @@ One of the following permissions is required to call this API. To learn more, in
 |:---------------------------------------|:--------------------------------------------|
 | Delegated (work or school account)     | Mail.Read, Calendars.Read, Files.Read.All, Sites.Read.All, ExternalItem.Read.All |
 | Delegated (personal Microsoft account) | Not supported. |
-| Application                            | Not supported. |
+| Application                            | Supported |
```
Comment (Lauragra) [inline api-reference/beta/api/search-query.md:28]: Instead of putting supported here, you need to list the names of the permissions that are required for application scenarios.

File: api-reference/beta/resources/includeContent.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
+---
+title: "includeContent enum type"
```
Comment (Lauragra) [inline api-reference/beta/resources/includeContent.md:2]: This is not the correct way to document enumerations. You can either define the enums in the enums.md topic or directly in the resource topic; we don't create separate individual topics for enums. Please see the guidance here: https://dev.azure.com/msazure/One/_wiki/wikis/Microsoft%20Graph%20Partners/211747/Enumerations

File: api-reference/beta/resources/sharepointonedriveoptions.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Provides the search content options when a search is performed using application permissions.
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|includeContent|[includeContent](includeContent.md)| Enum type which defined the available contents while search with application permission.|
```
[the remark this answers is not in the document]
Comment (yiwenwang-ms) [reply api-reference/beta/resources/sharepointonedriveoptions.md]: Hi @Lauragra , sorry for confused you here. Please check this CSDL
https://dev.azure.com/msazure/One/_git/AD-AggregatorService-Workloads/pullrequest/6715644?_a=files&path=/Workloads/Microsoft.MicrosoftSearch/override/schema-Prod-beta.csdl

The name is includeContent but the type named as searchContent.

    <ComplexType Name="sharePointOneDriveOptions" ags:WorkloadName="sharePointOneDriveOptions">
        <Property Name="includeContent" Type="microsoft.substrateSearch.searchContent"/>
   </ComplexType>
   
      <EnumType Name="searchContent" IsFlags="true">
        <Member Name="sharedContent" Value="1" />
        <Member Name="privateContent" Value="2" />
        <Member Name="unknownFutureValue" Value="4" />
      </EnumType>

File: api-reference/beta/resources/includeContent.md (hunk 11 above)
Comment (yiwenwang-ms) [reply api-reference/beta/resources/includeContent.md]: Thank you! I removed it and updated it followed guide.

File: api-reference/beta/resources/sharepointonedriveoptions.md [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+  "optionalProperties": [
+
+  ],
+  "@odata.type": "microsoft.graph.SharePointOneDriveOptions",
+  "baseType": null
+}-->
+
+```json
+{
+  "includeContent": "privateContent,sharedContent"
```
Comment (Lauragra) [inline api-reference/beta/resources/sharepointonedriveoptions.md:39]: The value here should be "string". For enums, we don't list the values in the JSON representation.

File: changelog/Microsoft.Search.json [hunk 14]
```diff
@@ -1199,6 +1199,40 @@
       "SubArea": ""
     },
     {
+      "ChangeList": [
+        {
+          "Id": "d7e73514-9dad-4c2f-8762-786788ac6984",
+          "ApiChange": "Enumeration",
+          "ChangedApiName": "includeContent",
+          "ChangeType": "Addition",
+          "Description": "Added the **includeContent** enumeration type.",
```
Comment (Lauragra) [inline changelog/Microsoft.Search.json:1208]: Please change this to searchContent.

File: changelog/Microsoft.Search.json [hunk 15]
```diff
@@ -1199,6 +1199,40 @@
       "SubArea": ""
     },
     {
+      "ChangeList": [
+        {
+          "Id": "d7e73514-9dad-4c2f-8762-786788ac6984",
```
Comment (Lauragra) [inline changelog/Microsoft.Search.json:1204]: Do you also need a ChangeList item to mention that you added support for application permissions to the search: query method?

File: changelog/Microsoft.Search.json (hunk 14 above)
Comment (yiwenwang-ms) [reply changelog/Microsoft.Search.json:1208]: Thanks! Done

File: api-reference/beta/resources/sharepointonedriveoptions.md [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Provides the search content options when a search is performed using application permissions.
+
+## Properties
+
+| Property     | Type        | Description |
+|:-------------|:------------|:------------|
+|includeContent|includeContent| The type of search content. The possible values are: `privateContent`, `sharedContent`. Read-only. |
```
[the remark this answers is not in the document]
Comment (yiwenwang-ms) [reply api-reference/beta/resources/sharepointonedriveoptions.md:22]: Thanks, done

File: api-reference/beta/resources/sharepointonedriveoptions.md (hunk 13 above)
Comment (yiwenwang-ms) [reply api-reference/beta/resources/sharepointonedriveoptions.md]: done

**Review by Lauragra — APPROVED** [review]
